Login with facebook

gregogerardi opened the pull request:
NOTA: para que el login con facebook funcione correctamente se debe generar una key propia de cada entorno de desarrollo, y agregarla a la lista de hash admitidas en la pagina de facebook developers de la aplicacion, instrucciones en https://developers.facebook.com/docs/facebook-login/android.

gregorio-gerardi committed:
avance 19-9

gregorio-gerardi committed:
Merge branch 'master' into login-with-facebook

gregorio-gerardi committed:
completo implementado con callbacks del callback manager

gregorio-gerardi committed:
login with facebook ready

NOTA: para que el login con facebook funcione correctamente se debe generar una key propia de cada entorno de desarrollo, y agregarla a la lista de hash admitidas en la pagina de facebook developers de la aplicacion, instrucciones en https://developers.facebook.com/docs/facebook-login/android

ghost commented:
I already noticed the four gradle generated files which shouldn't had been committed. And I forgot the setting.gradle file in the project root folder. I will change it on Monday.

gregorio-gerardi committed:
update gitignore and delete generated gradle files

gregorio-gerardi committed:
gradlew.bat added to gitignore

nacho-globant commented on FitTracker/app/src/main/java/com/pps/globant/fittracker/MainActivity.java:
```diff
@@ -35,4 +39,16 @@ protected void onPause() {
 
+    @Override
+    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
+        CallBackManagerProviderForFb.getCallbackManager().onActivityResult(requestCode, resultCode, data);
```
move to into presenter pls

nacho-globant commented on FitTracker/app/src/main/java/com/pps/globant/fittracker/mvp/model/LoginModel.java:
```diff
@@ -1,4 +1,17 @@
+
+    public LoginModel(Bus bus) {
+        this.bus = bus;
+        FacebookLoginProvider.registerCallback(bus);
```
move to a new method, and call since presenter. This simplify the test, and mantain the logic into presenter

nacho-globant commented on FitTracker/app/src/main/java/com/pps/globant/fittracker/mvp/presenter/LoginPresenter.java:
```diff
@@ -11,7 +15,31 @@
 
+    @Subscribe
+    public void onFetchingFbUserDataCompletedEvent(FacebookLoginProvider.FetchingFbUserDataCompletedEvent event) {
+        view.setLabelFb("name: " + event.fbUser.getName());
```
Use string format for that. See https://developer.android.com/reference/java/util/Formatter

nacho-globant commented on FitTracker/app/src/main/java/com/pps/globant/fittracker/mvp/presenter/LoginPresenter.java:
```diff
@@ -11,7 +15,31 @@
+
+    @Subscribe
+    public void onCantRetrieveAllTheFieldsRequestedsEvent(FacebookLoginProvider.CantRetrieveAllTheFieldsRequestedsEvent event) {
+        view.setLabelFb("error");
```
No harcode pls. Create a new item into strings.xml. Review the rest cases

SartiMau commented on FitTracker/app/src/main/java/com/pps/globant/fittracker/utils/FacebookLoginProvider.java:
```diff
@@ -61,9 +60,9 @@ public void onError(FacebookException exception) {
+        String name = null;
         try {
-            name=jsonObject.getString("name");
+            name = jsonObject.getString("name");
```
use constant, not hardcode.

SartiMau commented on FitTracker/app/src/main/java/com/pps/globant/fittracker/MainActivity.java:
```diff
@@ -18,9 +20,11 @@
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
+//todo move
```
this is util? if not, remove comment please

gregorio-gerardi committed:
code reviewed changes

delete coment in MainActivity.java
        CallBackManagerProviderForFb.getCallbackManager().onActivityResult(requestCode, resultCode, data);

 moved into pressenter

  public LoginModel(Bus bus) {
        this.bus = bus;
        FacebookLoginProvider.registerCallback(bus);

moved to another method, and called since pressenter only when a facebook login is required, so if not, we save memory used by the callbackmanager initialization

    @Subscribe
    public void onFetchingFbUserDataCompletedEvent(FacebookLoginProvider.FetchingFbUserDataCompletedEvent event) {
        view.setLabelFb("name: " + event.fbUser.getName());

String.Format used here

also added new items in strings.xml and replaced hardcoded strings by constants or references to strings.xml items.

marked some calss atributtes as final, and access modifiers of some classes atributtes and methods narrowed down.

capraber commented:
+1

vsmartinez commented on FitTracker/app/src/main/java/com/pps/globant/fittracker/MainActivity.java:
```diff
@@ -35,4 +36,16 @@ protected void onPause() {
 
+    @Override
+    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
+        presenter.redirectActivityResults(requestCode, resultCode, data);
```
call **super.onActivityResult** first

vsmartinez commented on FitTracker/app/src/main/java/com/pps/globant/fittracker/mvp/presenter/LoginPresenter.java:
```diff
@@ -11,7 +21,44 @@
 
+    @Subscribe
+    public void onFetchingFbUserDataCompletedEvent(FacebookLoginProvider.FetchingFbUserDataCompletedEvent event) {
+        Resources res= Objects.requireNonNull(view.getActivity()).getResources();
```
Exception not handled. Does null check is necessary?

vsmartinez commented on FitTracker/app/src/main/java/com/pps/globant/fittracker/mvp/presenter/LoginPresenter.java:
```diff
@@ -11,7 +21,44 @@
+    }
+
+    public void redirectActivityResults(int requestCode, int resultCode, Intent data) {
+        CallBackManagerProviderForFb.getCallbackManager().onActivityResult(requestCode, resultCode, data);
```
Singleton **CallBAckManagerProviderForFb** not needed, you can send a **CallbackManager** instance in the model constructor and use it here.

```  
callbackManager = CallbackManager.Factory.create();
presenter = new LoginPresenter(
                new LoginModel(
                        callbackManager,
                        BusProvider.getInstance()
                ),
                new LoginView(
                        this,
                        BusProvider.getInstance()
                )
        );
```

Then in the activity **onActivityResult** do:

```
 @Override
    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
        super.onActivityResult(requestCode, resultCode, data);
        callbackManager.onActivityResult(requestCode, resultCode, data);
    }
```

gregogerardi replied:
about the second comment... nacho told me to move the redirection to the callbackmanager of the activity result to the presenter (or that's what i understood), see https://github.com/priettt/FitTracker/pull/2#discussion_r219838068. I already undid this change to independise the presenter from the activity's life cicle, just a comment

gregorio-gerardi committed:
changes post revision

singleton replaced with a dependence, activity's results no longer redirected to the callback manager through the presenter, null's inspection deleted
